### ISSUE TITLE
MD057 consider excluding special uri schemes

### ISSUE THREAD

**Issue by upekkha** [rvben/rumdl#192]
The relative link checker would benefit from excluding various URI schemes, where existence can not easily be checked.

````markdown
# MD057 false positives with special uri schemes

* [Link with file uri](file:///path/to/file)
* [Link with smb uri](smb://example.com/path/to/share)
* [Link with macappstores uri](macappstores://apps.apple.com/)
* and so on
````

```
3:3: [MD057] Relative link 'file:///path/to/file' does not exist
4:3: [MD057] Relative link 'smb://example.com/path/to/share' does not exist
5:3: [MD057] Relative link 'macappstores://apps.apple.com/' does not exist
```

**Comment by rvben** [comment]
Hi @upekkha, thanks for the issue. I recognise the need. Probably ignoring these would make most sense?

**Comment by rvben** [comment]
MD057 was using a hardcoded list of common protocols (http, https, ftp) to identify external URLs that should be skipped. Any URI scheme not in that list would be treated as a relative file path and checked for existence, causing false positives.

This is now resolved.

Thanks @upekkha for raising this issue. 🚀
